includes/Modules/Search_Console.php

```diff
@@ -52,6 +52,27 @@ public function register() {
 
 		$this->register_screen_hook();
 
+		// Detect and store Search Console property when receiving token for the first time.
+		add_action(
+			'googlesitekit_authorize_user',
+			function() {
+				// Bail if there is one set already.
+				$property_id = $this->get_property_id();
+				if ( $property_id ) {
+					return;
+				}
+
+				$property_id = $this->detect_property_id();
+				if ( ! $property_id ) {
+					return;
+				}
```

Comment:
If no property ID is detected, should it not still save that in the settings (potentially removing one that was previously saved)?

Reply:
Good point, I'll update that.

Reply:
Having second thoughts, I don't think we should save an empty value here as I can see it causing more harm than help. If a user authenticates and e.g. the request fails unexpectedly, it shouldn't cause the existing SC property to be wiped. That would result in a worse experience than just keeping the previously stored one (which would likely be the same anyway).

Reply:
This could be simplified into a single conditional:
```php
$property_id = $this->get_property_id() ?: $this->detect_property_id();
if ( ! $property_id ) {
    return;
}
```